aws/resource_aws_sns_topic_subscription.go

```diff
@@ -249,6 +249,10 @@ func subscribeToSNSTopic(d *schema.ResourceData, snsconn *sns.SNS) (output *sns.
 				fmt.Errorf("Endpoint (%s) did not autoconfirm the subscription for topic %s", endpoint, topic_arn))
 		})
 
+		if isResourceTimeoutError(err) {
+			_, err = findSubscriptionByNonID(d, snsconn)
```

Comment:
The logic above here seems to be modifying the `output.SubscriptionArn` if its found:

```go
			subscription, err := findSubscriptionByNonID(d, snsconn)

			if subscription != nil {
				output.SubscriptionArn = subscription.SubscriptionArn
				return nil
			}
```

This functionality should be replicated here as well (without the `return nil`):

```suggestion
			var subscription *sns.Subscription
			subscription, err = findSubscriptionByNonID(d, snsconn)
			
			if subscription != nil {
				output.SubscriptionArn = subscription.SubscriptionArn
			}
```